frontend/src/app/main/data/workspace/shapes.cljs

```diff
@@ -131,141 +131,142 @@
-
-            components-to-delete
-            (if components-v2
-              (reduce (fn [components id]
-                        (let [shape (get objects id)]
-                          (if (and (= (:component-file shape) file-id) ;; Main instances should exist only in local file
-                                   (:main-instance? shape))            ;; but check anyway
-                            (conj components (:component-id shape))
-                            components)))
-                      []
-                      (into ids all-children))
-              [])
-
-            changes (-> (pcb/empty-changes it page-id)
-                        (pcb/with-page page)
-                        (pcb/with-objects objects)
-                        (pcb/with-library-data file)
-                        (pcb/set-page-option :guides guides))
-
-            changes (reduce (fn [changes component-id]
-                              ;; It's important to delete the component before the main instance, because we
-                              ;; need to store the instance position if we want to restore it later.
-                              (pcb/delete-component changes component-id components-v2))
-                            changes
-                            components-to-delete)
-
-            changes (-> changes
-                        (pcb/remove-objects all-children)
-                        (pcb/remove-objects ids)
-                        (pcb/remove-objects empty-parents)
-                        (pcb/resize-parents all-parents)
-                        (pcb/update-shapes groups-to-unmask
-                                           (fn [shape]
-                                             (assoc shape :masked-group? false)))
-                        (pcb/update-shapes (map :id interacting-shapes)
-                                           (fn [shape]
-                                             (d/update-when shape :interactions
-                                                            (fn [interactions]
-                                                              (into []
-                                                                    (remove #(and (ctsi/has-destination %)
-                                                                                  (contains? ids (:destination %))))
-                                                                    interactions)))))
-                        (cond-> (seq starting-flows)
-                          (pcb/update-page-option :flows (fn [flows]
-                                                           (->> (map :id starting-flows)
-                                                                (reduce ctp/remove-flow flows))))))]
-
-        (rx/of (dc/detach-comment-thread ids)
-               (dwsl/update-layout-positions all-parents)
-               (dch/commit-changes changes))))))
+  ([ids] (delete-shapes nil ids))
```

Comment:
No changes here. Only added a second arity with can specify the page-id, if different to :current-page-id